### ISSUE TITLE
Memory usage constantly grows

### ISSUE THREAD

**Issue by zet4** [pingcap/tikv#1471]
Please answer these questions before submitting your issue. Thanks!

1. What version of Rust are you using (`rustc --version`)?
Whichever latest docker image comes with

2. What operating system and processor architecture are you using?
Linux 64bit

3. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
Started full 1tidb 3pd 3tikv cluster using docker-compose
The cluster is running idly, nothing is using it.

4. What did you expect to see?
Memory usage of all TIKV instances to remain at ~150-200mb

5. What did you see instead?
Memory usage went from ~150mb on day one to 500mb+ on day three.

**Comment by siddontang** [comment]
Hi @zet4 

> The cluster is running idly, nothing is using it.

Is is mean that you do nothing after the cluster starts, and then TiKV uses 500MB+ after two days? 

/cc @zhangjinpeng1987

**Comment by zet4** [comment]
Yeah, currently I have 3 instances of PD and 3 TiKV instances, started at `2017-01-04 17:39:52`, the cluster is not being used in any way (I wanted to migrate gitea from PG to TiDB but thats later if this can be resolved) and each instance of TiKV is using ~700mb right now.

**Comment by zet4** [comment]
My portainer dashboard showing 1 tikv instance right now:
![image](https://my.foxgirlsare.sexy/uploads/17/01/07/1o6U9AYB.png)

**Comment by siddontang** [comment]
Thanks @zet4 

We will see it later.

**Comment by siddontang** [comment]
/cc @zhangjinpeng1987 

We should first see whether this memory is occupied by RocksDB. we can dump RocksDB block cache to metric.

**Comment by zhangjinpeng1987** [comment]
@zet4 May I take a look at your config file of TiKV`tikv.toml`?

**Comment by zet4** [comment]
As I said before, it's using Docker Compose file that can be found [here](
https://github.com/pingcap/docs/blob/master/op-guide/docker-compose.md).

2017-01-14 6:36 GMT+02:00 zhangjinpeng1987 <notifications@github.com>:

**Comment by zhangjinpeng1987** [comment]
@zet4 We use rocksdb's column families feature, there are four different column families in TiKV, every column family has independent memtable(default value 128MB) and independent block-cache.   Though cluster is not being used, but messages(raft) between TiKVs will never stop, raft messages will write to rocksdb(this will increase memory usage of memtable) and will be read(blocks will fill into block-cache, and the memory usage of block-cache will increase) later.
We will continue to track this issue, thanks.
